wearApp/src/main/java/dev/johnoreilly/confetti/wear/decompose/SwipeToDismissBox.kt

```diff
@@ -0,0 +1,178 @@
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.SaveableStateHolder
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.saveable.rememberSaveableStateHolder
+import androidx.compose.ui.Modifier
+import androidx.wear.compose.foundation.ExperimentalWearFoundationApi
+import androidx.wear.compose.foundation.OnFocusChange
+import androidx.wear.compose.foundation.lazy.ScalingLazyListState
+import androidx.wear.compose.material.PositionIndicator
+import androidx.wear.compose.material.Scaffold
+import androidx.wear.compose.material.SwipeToDismissBox
+import androidx.wear.compose.material.SwipeToDismissKeys
+import androidx.wear.compose.material.TimeText
+import com.arkivanov.decompose.Child
+import com.arkivanov.decompose.extensions.compose.jetpack.subscribeAsState
+import com.arkivanov.decompose.router.stack.ChildStack
+import com.arkivanov.decompose.value.Value
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults
+import com.google.android.horologist.compose.layout.ScalingLazyColumnState
+import com.google.android.horologist.compose.layout.scrollAway
+
+/**
+ * Displays the [ChildStack] in [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ *
+ * @param stack a [ChildStack] to be displayed.
+ * @param onDismissed called when the swipe to dismiss gesture has completed, allows popping the stack.
+ * See [StackNavigator#pop][com.arkivanov.decompose.router.stack.pop].
+ * @param modifier a [Modifier] to be applied to the underlying
+ * [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ * @param content a Composable slot displaying a [Child][Child.Created].
+ */
+@Composable
+fun <C : Any, T : Any> SwipeToDismissBox(
+    stack: Value<ChildStack<C, T>>,
+    onDismissed: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable SwipeToDismissBoxScope.(child: Child.Created<C, T>) -> Unit,
+) {
+    val state = stack.subscribeAsState()
+
+    SwipeToDismissBox(
+        stack = state.value,
+        onDismissed = onDismissed,
+        modifier = modifier,
+        content = content,
+    )
+}
+
+class SwipeToDismissBoxScope {
+    private val _scalingLazyColumnState = mutableStateOf<ScalingLazyColumnState?>(null)
+
+    val scrollableState: ScalingLazyColumnState?
+        get() = _scalingLazyColumnState.value
+
+    @Composable
+    fun createScalingLazyColumnState(
+        factory: ScalingLazyColumnState.Factory = ScalingLazyColumnDefaults.belowTimeText()
+    ): ScalingLazyColumnState {
+        val scalingLazyColumnState = factory.create()
+        _scalingLazyColumnState.value = scalingLazyColumnState
+
+        scalingLazyColumnState.state = rememberSaveable(saver = ScalingLazyListState.Saver) {
```

Comment:
@arkivanov I just added this so that the LazyColumn position is retained when you swipe back.

Reply:
Interesting, I thought that the existing `dev.johnoreilly.confetti.wear.navigation.SwipeToDismissBox` (which you removed) should have been handling this already. E.g. it wraps child screens with `SaveableStateProvider` which should keep the state.

Reply:
It does work. The issue here is the app level scaffold needs access to the screen level scroll state.

Not sure how else to plumb it through.

No issue, just complication of wear with time text at top of the app